<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Loops (Good)</span></div>

## 1 Interrupting the flow

## 2 List comprehension!

### 2.1 Basic syntax

In [1]:

# Your code here


### 2.2 List comprehension with conditions

In [2]:

# Your code here


## 3 Other useful stuff

### 3.1 for with unpacking

In [3]:

# Your code here


### 3.2 for with a zip()

In [4]:

# Your code here


### 3.3 for with dictionaries

In [5]:

# Your code here


## Exercise 1 :  Make me an odd list

In [13]:
%%timeit
a = []
for i in range(10):
    if i % 2 == 0:
        continue
    a.append(i)

2.82 µs ± 696 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)


## Exercise 2 :  Make me another odd list

In [12]:
%%timeit
a = [i for i in range(1,10,2)]

1.41 µs ± 121 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


## Exercise 3 :  Time me!

Exercise 1: 2.82 µs ± 696 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
Exercise 2: 1.41 µs ± 121 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
List comprehension is faster.